### PR TITLE
Feature/smart vu plugin

### PR DESCRIPTION
**Pull request by lgdacunha**
Added Solidigm plug-in for additional SMART fields.

### CODE REVIEW

**Comment by lgdacunha** [comment]
Hello @igaw,
I am a Solidigm engineer. Can you give me some pointers on what steps should I follow to have this code reviewed?

**Comment by igaw** [comment]
Will do when I am back from my holiday.

File: plugins/solidigm/solidigm-nvme.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#undef CMD_INC_FILE
```
Comment (igaw) [inline plugins/solidigm/solidigm-nvme.h:1]: Please add a `SPDX-License-Identifier: GPL-2.0-or-later` and copyright statement here. No need for the old style copyright header. (FWIW, I want to follow the kernel and have just the SPDX identifier at the beginning of each file)

File: plugins/solidigm/solidigm-smart.c [hunk 2]
```diff
@@ -0,0 +1,260 @@
+#include <fcntl.h>
```
Comment (igaw) [inline plugins/solidigm/solidigm-smart.c:1]: Please add a SPDX-License-Identifier: GPL-2.0-or-later and copyright statement here too.

File: plugins/solidigm/solidigm-smart.c [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+		return;
+	default:
+		printf("%"PRIu64"\n", int48_to_long(item->raw));
+	}
+}
+
+static void smart_log_item_add_json(smart_log_item_t *item, struct json_object *dev_stats)
+{
+	if (!item->id) 
+	{
```
Comment (igaw) [inline plugins/solidigm/solidigm-smart.c:137]: see above about formatting rules.

File: plugins/solidigm/solidigm-smart.c [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+		return;
+	}
+	struct json_object *entry_stats = json_create_object();
```
Comment (igaw) [inline plugins/solidigm/solidigm-smart.c:140]: Such deceleration should got first.

File: plugins/solidigm/solidigm-smart.c [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+	json_object_add_value_object(dev_stats, id_to_name(item->id), entry_stats);
+}
+
+static void vu_smart_log_show_json(vu_smart_log_t *payload, unsigned int nsid, const char *devname)
+{
+	struct json_object *dev_stats = json_create_object();
+
+	smart_log_item_t *item = payload->item;
+	for (int i = 0; i < VU_SMART_MAX_ITEMS; i++)
+	{
```
Comment (igaw) [inline plugins/solidigm/solidigm-smart.c:166]: this bracket should go to up `for () {`

File: plugins/solidigm/solidigm-smart.c [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+	}
+	json_object_add_value_object(dev_stats, id_to_name(item->id), entry_stats);
+}
+
+static void vu_smart_log_show_json(vu_smart_log_t *payload, unsigned int nsid, const char *devname)
+{
+	struct json_object *dev_stats = json_create_object();
+
+	smart_log_item_t *item = payload->item;
+	for (int i = 0; i < VU_SMART_MAX_ITEMS; i++)
```
Comment (igaw) [inline plugins/solidigm/solidigm-smart.c:165]: one empty line after declarations

File: plugins/solidigm/solidigm-smart.c [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+	json_free_object(root);
+}
+
+static void vu_smart_log_show(vu_smart_log_t *payload, unsigned int nsid, const char *devname)
+{
+	printf("Additional Smart Log for NVME device:%s namespace-id:%x\n",
+		devname, nsid);
+	printf("ID             KEY                                 Normalized     Raw\n");
+
+	smart_log_item_t *item = payload->item;
```
Comment (igaw) [inline plugins/solidigm/solidigm-smart.c:184]: declarations go first.

**Review by igaw — CHANGES_REQUESTED** [review]
- Missing copyright statement (SPDX)
- formatting/coding style

File: plugins/solidigm/solidigm-nvme.h (hunk 1 above)
Comment (lgdacunha) [reply plugins/solidigm/solidigm-nvme.h:22]: Done!

**Comment by igaw** [comment]
I've rebased it to the current master and did some small style changes.

**Review by igaw — APPROVED** [review]

**Pull request merged**